common/stats/status_server.h

```diff
@@ -59,11 +60,18 @@ class StatusServer {
    * the function in the form of a vector of [key, value] pairs
    * (arrays can be passed in as delimited values) and returns a
    * std::string.
+   * @param extra_stats_endpoints For endpoints in this list, their string
+   * result will be appended to the regular stats.txt endpoint result. Note:
+   * These endpoints' format need to confirm to the existing stats.txt format,
+   * so that the same stats collector can consume it. Also they cannot use keys
+   * including "counters" and "metrics".
```

Comment:
I think it only allows "counter" type stats? Because the last section of the original "stats.txt" endpoint is for counters. We'd document it also.